### DISCUSSION TITLE
List spacing breaks with multi-level lists

### DISCUSSION THREAD

**Discussion by arozbiz** [typst/typst#4395] [General]
### Description

Non-tight list spacing breaks with multi-level lists. Examples below. I would imagine the expected behavior would be for all the bullets to be equally distributed vertically, whether or not they are second-level indented.

This works as expected:
```
#set page(height: 150pt, width: 200pt)
#set text(size: 10pt)

#set list(spacing: 1fr)
- a

- b

- c

- d

- e
```
![image](https://github.com/typst/typst/assets/4499657/013dc0d0-e3e1-479e-844c-77a7c06f7775)

This has some issues (the difference between a and b is different than the difference between b and c):
```
#set page(height: 150pt, width: 200pt)
#set text(size: 10pt)

#set list(spacing: 1fr)
- a

  - b

- c

- d

- e
```
![image](https://github.com/typst/typst/assets/4499657/8ca6fb83-f18b-4446-81c3-eab691cfa5ad)

This one completely breaks:
```
#set page(height: 150pt, width: 200pt)
#set text(size: 10pt)

#set list(spacing: 1fr)
- a

  - b

  - c

- d

- e
```
![image](https://github.com/typst/typst/assets/4499657/240799e9-aa98-4b6a-a3f7-dc28a8053e52)

### Operating system

Linux

### Typst version

- [X] I am using the latest version of Typst

**Comment by arozbiz** [comment]
Any progress on this?

**Comment by laurmaedje** [comment]
No progress so far

**Comment by laurmaedje** [comment]
Okay, I looked at this again.

You can achieve this by also configuring the `above`/`below` spacing of the list's block. Fundamentally, list spacing is only between the items. So something like this:
```typ
#set page(height: 150pt, width: 200pt)
#set text(size: 10pt)
#set list(spacing: 20pt)
#show list: set block(above: 20pt)

- a

  - b

  - c
  
- d

- e
```

Note that this doesn't work with tight lists directly after paragraph because they attach themselves to the paragraph.

**Comment by arozbiz** [comment]
Thanks but I was hoping for a solution that would automatically distribute bullets across the page, no matter how many there are. For reference, [this](https://tex.stackexchange.com/questions/369504/beamer-vertically-stretching-level-1-list-items-in-a-nested-list-environment#369597) is how I currently implement this in LaTeX/Beamer.
